modules/openapi-generator/src/main/resources/aspnetcore/2.1/Startup.mustache

```diff
@@ -42,11 +42,11 @@ namespace {{packageName}}
         {
             // Add framework services.
             services
-                .AddMvc()
+                .AddMvc({{^useDefaultRoutng}}opts => opts.EnableEndpointRouting = false{{/useDefaultRoutng}})
                 {{#compatibilityVersion}}
                 .SetCompatibilityVersion(CompatibilityVersion.{{compatibilityVersion}})
                 {{/compatibilityVersion}}
-                .AddJsonOptions(opts =>
+                .{{#useNewtonsoft}}AddNewtonsoftJson{{/useNewtonsoft}}{{^useNewtonsoft}}AddJsonOptions{{/useNewtonsoft}}(opts =>
```

Comment:
Variable `useNewtonsoft` will also need to be evaluated in `controller.mustache` and `model.mustache`. Currently, the artifact won't be loaded, but controllers and models are still tightly coupled to json serialize/deserialize using the library.

Reply:
see the earlier comment about setting the value of useNewtonsoft.

Reply:
I'm not sure I follow.

To put the concern a different way, as a user if I set `useNewtonSoft=false`, and the controller/model generation still includes NewtonSoft imports, I'd consider that a bug. Maybe the variable name is what's misleading here.

Reply:
Indeed, since the models are generated with `Netwonsoft` in mind it is required that in `3.x` it is included via the package dependency and for `2.x` is there as part of core fw.